### ISSUE TITLE
ActionCable + Ruby 2.5.0-rc1 Argument Error

### ISSUE THREAD

**Issue by Petercopter** [rails/rails#31517]
### Steps to reproduce

Install Ruby 2.5.0-rc1 
Load a page with ActionCable listening

### Expected behavior
Log should be free of errors

### Actual behavior
 wrong number of arguments (given 283856384, expected 0) 😸 
```
#<Thread:0x00007fbb44c01b90@/Users/petercopter/.rbenv/versions/2.5.0-rc1/lib/ruby/gems/2.5.0/gems/actioncable-5.1.4/lib/action_cable/connection/stream_event_loop.rb:73 run> terminated with exception (report_on_exception is true):
Traceback (most recent call last):
	5: from /Users/petercopter/.rbenv/versions/2.5.0-rc1/lib/ruby/gems/2.5.0/gems/actioncable-5.1.4/lib/action_cable/connection/stream_event_loop.rb:73:in `block (2 levels) in spawn'
	4: from /Users/petercopter/.rbenv/versions/2.5.0-rc1/lib/ruby/gems/2.5.0/gems/actioncable-5.1.4/lib/action_cable/connection/stream_event_loop.rb:84:in `run'
	3: from /Users/petercopter/.rbenv/versions/2.5.0-rc1/lib/ruby/gems/2.5.0/gems/actioncable-5.1.4/lib/action_cable/connection/stream_event_loop.rb:84:in `loop'
	2: from /Users/petercopter/.rbenv/versions/2.5.0-rc1/lib/ruby/gems/2.5.0/gems/actioncable-5.1.4/lib/action_cable/connection/stream_event_loop.rb:94:in `block in run'
	1: from /Users/petercopter/.rbenv/versions/2.5.0-rc1/lib/ruby/gems/2.5.0/gems/actioncable-5.1.4/lib/action_cable/connection/stream_event_loop.rb:94:in `select'
/Users/petercopter/.rbenv/versions/2.5.0-rc1/lib/ruby/gems/2.5.0/gems/actioncable-5.1.4/lib/action_cable/connection/stream_event_loop.rb:94:in `lock': wrong number of arguments (given 283856384, expected 0) (ArgumentError)
WebSocket error occurred: Broken pipe
```

### System configuration
Rails 5.1.4
Ruby 2.5.0-rc1

**Comment by eileencodes** [comment]
@Petercopter thanks for the report. Can you provide a new application on Rails 5.1.4 or master with Ruby 2.5.0 that reproduces this error? That will help us track this down. Thanks!

**Comment by Petercopter** [comment]
@eileencodes Thanks for looking into this. I've setup a new application here:
https://github.com/Petercopter/ruby_2_5_test

I used the basic ActionCable instructions to get up and running, and it appears to be pretty similar to what I have in production. I confirmed that things were working with 2.4.2 before switching over to 2.5.0, and luckily the error is readily apparent.

If you pull the application, just start server and visits `/posts`, you will see it. Let me know what else I can do, I'm happy to be helping. 

```
Completed 200 OK in 251ms (Views: 241.2ms | ActiveRecord: 0.4ms)

Started GET "/cable" for 127.0.0.1 at 2017-12-20 14:19:51 -0800
Started GET "/cable/" [WebSocket] for 127.0.0.1 at 2017-12-20 14:19:51 -0800
Successfully upgraded to WebSocket (REQUEST_METHOD: GET, HTTP_CONNECTION: Upgrade, HTTP_UPGRADE: websocket)
#<Thread:0x00007fbb8cb5d9d0@/Users/petercopter/.rbenv/versions/2.5.0-rc1/lib/ruby/gems/2.5.0/gems/actioncable-5.1.4/lib/action_cable/connection/stream_event_loop.rb:73 run> terminated with exception (report_on_exception is true):
Traceback (most recent call last):
	6: from /Users/petercopter/.rbenv/versions/2.5.0-rc1/lib/ruby/gems/2.5.0/gems/actioncable-5.1.4/lib/action_cable/connection/stream_event_loop.rb:73:in `block (2 levels) in spawn'
	5: from /Users/petercopter/.rbenv/versions/2.5.0-rc1/lib/ruby/gems/2.5.0/gems/actioncable-5.1.4/lib/action_cable/connection/stream_event_loop.rb:84:in `run'
	4: from /Users/petercopter/.rbenv/versions/2.5.0-rc1/lib/ruby/gems/2.5.0/gems/actioncable-5.1.4/lib/action_cable/connection/stream_event_loop.rb:84:in `loop'
	3: from /Users/petercopter/.rbenv/versions/2.5.0-rc1/lib/ruby/gems/2.5.0/gems/actioncable-5.1.4/lib/action_cable/connection/stream_event_loop.rb:91:in `block in run'
	2: from /Users/petercopter/.rbenv/versions/2.5.0-rc1/lib/ruby/gems/2.5.0/gems/actioncable-5.1.4/lib/action_cable/connection/stream_event_loop.rb:29:in `block in attach'
	1: from /Users/petercopter/.rbenv/versions/2.5.0-rc1/lib/ruby/gems/2.5.0/gems/actioncable-5.1.4/lib/action_cable/connection/stream_event_loop.rb:29:in `register'
/Users/petercopter/.rbenv/versions/2.5.0-rc1/lib/ruby/gems/2.5.0/gems/actioncable-5.1.4/lib/action_cable/connection/stream_event_loop.rb:29:in `lock': wrong number of arguments (given 4, expected 0) (ArgumentError)
```

**Comment by y-yagi** [comment]
Seems issue of nio4r. And already fixed with v2.2.0.
https://github.com/socketry/nio4r/pull/179
https://github.com/socketry/nio4r/blob/master/CHANGES.md#220-2017-12-27

Please update nio4r gem. Thanks!

**Comment by Petercopter** [comment]
@y-yagi Awesome, thanks! :thumbsup: :tada:
